lte/gateway/c/oai/tasks/sgw/sgw_handlers.c

```diff
@@ -2368,8 +2371,213 @@ int sgw_handle_nw_initiated_actv_bearer_rsp(
+      eps_bearer_ctxt_p->s_gw_teid_S1u_S12_S4_up;
+    sgi_delete_end_point_request.eps_bearer_id = ebi;
+    sgi_delete_end_point_request.pdn_type =
+      spgw_ctxt->sgw_eps_bearer_context_information.saved_message.pdn_type;
+    memcpy(
+      &sgi_delete_end_point_request.paa,
+      &eps_bearer_ctxt_p->paa,
+      sizeof(paa_t));
+
+    sgw_handle_sgi_endpoint_deleted(&sgi_delete_end_point_request);
+
+    sgw_cm_remove_eps_bearer_entry(
+      &spgw_ctxt->sgw_eps_bearer_context_information.pdn_connection,
+      ebi);
+
+    sgw_cm_remove_bearer_context_information(
+      s11_pcrf_ded_bearer_deactv_rsp->s_gw_teid_s11_s4);
+
+    sgw_cm_remove_s11_tunnel(s11_pcrf_ded_bearer_deactv_rsp->s_gw_teid_s11_s4);
+  } else {
+      //Remove the dedicated bearer/s context
+      for (i = 0; i < no_of_bearers; i++) {
+        eps_bearer_ctxt_p = sgw_cm_get_eps_bearer_entry(
+          &spgw_ctxt->sgw_eps_bearer_context_information.pdn_connection,
+          s11_pcrf_ded_bearer_deactv_rsp->bearer_contexts.bearer_contexts[i].
+          eps_bearer_id);
+        if (eps_bearer_ctxt_p) {
+          ebi = s11_pcrf_ded_bearer_deactv_rsp->bearer_contexts.
+            bearer_contexts[i]
+            .eps_bearer_id;
+          OAILOG_INFO(
+            LOG_SPGW_APP,
+            "Removed bearer context for (ebi = %d)\n", ebi);
+          rc = gtp_tunnel_ops->del_tunnel(
+            eps_bearer_ctxt_p->paa.ipv4_address,
+            eps_bearer_ctxt_p->s_gw_teid_S1u_S12_S4_up,
+            eps_bearer_ctxt_p->enb_teid_S1u,
+            NULL);
+          if (rc < 0) {
+            OAILOG_ERROR(
+            LOG_SPGW_APP,
+            "ERROR in deleting TUNNEL " TEID_FMT
+            " (eNB) <-> (SGW) " TEID_FMT "\n",
+            eps_bearer_ctxt_p->enb_teid_S1u,
+            eps_bearer_ctxt_p->s_gw_teid_S1u_S12_S4_up);
+          }
+
+          sgw_free_sgw_eps_bearer_context(
+            &spgw_ctxt->sgw_eps_bearer_context_information.
+            pdn_connection.sgw_eps_bearers_array[ebi]);
+          break;
+        }
+      }
+    }
+  //Send DEACTIVATE_DEDICATED_BEARER_RSP to PGW
+  MessageDef *message_p = NULL;
+  message_p =
+    itti_alloc_new_message(TASK_PGW_APP,
+    S5_NW_INITIATED_DEACTIVATE_BEARER_RESP);
+  if (message_p == NULL) {
+    OAILOG_ERROR(
+      LOG_MME_APP,
+      "itti_alloc_new_message failed for S5_NW_INITIATED_DEACTV_BEARER_RESP\n");
+    OAILOG_FUNC_RETURN(LOG_SPGW_APP, RETURNerror);
+  }
+  itti_s5_nw_init_deactv_bearer_rsp_t *deact_ded_bearer_rsp =
+    &message_p->ittiMsg.s5_nw_init_deactv_bearer_response;
+  deact_ded_bearer_rsp->no_of_bearers =
+    s11_pcrf_ded_bearer_deactv_rsp->bearer_contexts.num_bearer_context;
+
+  for (i = 0; i < deact_ded_bearer_rsp->no_of_bearers; i++) {
+    //EBI
+    deact_ded_bearer_rsp->ebi[i] = ebi;
+    //Cause
+    deact_ded_bearer_rsp->cause.cause_value =
+      s11_pcrf_ded_bearer_deactv_rsp->bearer_contexts.
+      bearer_contexts[i].cause.cause_value;
+
+    OAILOG_INFO(
```

Comment:
take log outside loop and print list of EBIs instead

Reply:
done